### DISCUSSION TITLE
Exercício de JS

### DISCUSSION THREAD

**Discussion by lucastorresbr** [frontendbr/forum#1867] [Perguntas e Respostas]
Olá!

Estou fazendo exercícios em JS e estou com algumas dúvidas no meu código.

Segue pergunta:

Programar uma função filme que recebe três arrays com os nomes de personagens, filmes e ano de estréia no cinema. 
A função deve receber também um valor de id escolhido pelo usuário com intervalo de 1 até o tamanho de um dos arrays fornecidos e retornar uma frase com o seguinte modelo: 

"personagem é um personagem do filme filme que estreou no cinema em lançamento." 

Exemplo:
Se o valor de id for inválido, a função deve retornar a frase "Essa não é uma opção válida."

var personagens = ["Hermione", "Trinity", "Leia"]
var filmes = ["Harry Potter", "Matrix", "Star wars"]
var lancamentos = [2001, 1999, 1977]
var id = 3

filme(personagens, filmes, lancamentos, id)  
// retorna a string "Leia é um personagem do filme Star wars que estreou no cinema em 1977."

MEU CÓDIGO:

function filme(personagens, filmes, lancamentos, id)
{
var personagens = ["Hermione", "Trinity", "Leia"]
var filmes = ["Harry Potter", "Matrix", "Star wars"]
var lancamentos = [2001, 1999, 1977]
var id = 2

return filme (personagens, "é um personagem do" ,filme, "filme que estreou no cinema em ",lancamento)
}

filme(personagens, filmes, lancamentos, id)

**Comment by LuisMatheus-dev** [comment]
Olá @lucastorresbr Tudo bem ?
O ideal neste caso seriam objetos ao inves de arrays , creio que ficaria mais consivo e não se limitaria ao ordem dos dados (index) ou quantidade. Porem como foi passado segue a minha resolução


```
personagens = ["Hermione", "Trinity", "Leia"];
filmes = ["Harry Potter", "Matrix", "Star wars"];
lancamentos = [2001, 1999, 1977];
id = 3;


function filme(id) {
    // Listas de Informação referente ao personagem
    let lanc, person, filme
    
    //Informações do personagem
    lanc = lancamentos[id] 
    person = personagens[id];
    filme = filmes[id];

    // Se o valor do Id não existir (undefined)
    if (personagens[id] === undefined) {
        return `Esta não é uma opção valida`;

    } else {
        return `${person} é um personagem do filme ${filme}, que estreou em ${lanc}`;
    }
}

console.log(filme(0));

```
